airbyte-integrations/connectors/destination-vectara/destination_vectara/client.py

```diff
@@ -0,0 +1,222 @@
+        }
+
+        request_time = datetime.datetime.now().timestamp()
+        response = requests.request(method="POST", url=token_endpoint, headers=headers, data=data)
+        response_json = response.json()
+
+        self.jwt_token = response_json.get("access_token")
+        self.jwt_token_expires_ts = request_time + response_json.get("expires_in")
+        return self.jwt_token
+    
+    @backoff.on_exception(backoff.expo, requests.exceptions.RequestException, max_tries=5, giveup=user_error)
+    def _request(
+            self, endpoint: str, 
+            http_method: str = "POST", 
+            params: Mapping[str, Any] = None, 
+            data: Mapping[str, Any] = None
+        ):
+        
+        url = f"{self.BASE_URL}/{endpoint}"
+
+        current_ts = datetime.datetime.now().timestamp()
+        if self.jwt_token_expires_ts - current_ts <= 60:
+            self._get_jwt_token()
+
+        headers = {
+            "Content-Type": "application/json",
+            "Accept": "application/json", 
+            "Authorization": f"Bearer {self.jwt_token}",
+            "customer-id": self.customer_id,
+            "X-source": "airbyte"
+            }
+
+        response = requests.request(method=http_method, url=url, headers=headers, params=params, data=json.dumps(data))
+        response.raise_for_status()
+        return response.json()
+
+    def delete_doc_by_metadata(self, metadata_field_name, metadata_field_values):
+        document_ids = []
+        for value in metadata_field_values:
+            data = {
+                "query": [
+                    {
+                        "query": "", 
+                        "numResults": 100,
+                        "corpusKey": [
+                            {
+                                "customerId": self.customer_id,
+                                "corpusId": self.corpus_id,
+                                "metadataFilter": f"doc.{metadata_field_name} = '{value}'"
+                            }
+                        ]
+                    }
+                ]
+            }
+            query_documents_response = self._request(
+                endpoint="query",
+                data = data
+            )
+            document_ids.extend([document.get("id") for document in query_documents_response.get("responseSet")[0].get("document")])
+        return self.delete_docs_by_id(document_ids=document_ids)
+    
+    def delete_docs_by_id(self, document_ids):
+        documents_not_deleted = []
+        for document_id in document_ids:
+            delete_document_response = self._request(
+                endpoint="delete-doc",
+                data={
+                    "customerId": self.customer_id, 
+                    "corpusId": self.corpus_id,
+                    "documentId": document_id
+                    }
+                )
+            # TODO whether this is needed?
+            if delete_document_response:
```

Comment:
This stuff doesn't seem to be necessary

Reply:
Yes agree. Was from original code. Will remove